tempodb/pool/pool.go

```diff
@@ -29,11 +29,10 @@ var (
 	})
 )
 
-type JobFunc func(ctx context.Context, payload interface{}) ([]byte, string, error)
+type JobFunc func(ctx context.Context, payload interface{}) (interface{}, error)
 
 type result struct {
-	data []byte
-	enc  string
+	data interface{}
```

Comment:
is there any reason to use an empty interface over *tempopb.Trace?

Reply:
If we wanted to keep the pool around for any other activities.   Following comments on above, let's remove it in a next PR.